ddtrace/contrib/botocore/patch.py

```diff
@@ -467,10 +496,52 @@ def patched_api_call(original_func, instance, args, kwargs):
         span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.botocore.get_analytics_sample_rate())
 
         try:
-            result = original_func(*args, **kwargs)
-            _set_response_metadata_tags(span, result)
+            if endpoint_name == "sqs" and operation == "ReceiveMessage" and config._data_streams_enabled:
+                queue_url = params["QueueUrl"]
+                queue_name = queue_url[queue_url.rfind("/") + 1 :]
+
+                if "MessageAttributeNames" not in params:
+                    params.update({"MessageAttributeNames": ["_datadog"]})
+                elif "_datadog" not in params["MessageAttributeNames"]:
+                    params.update({"MessageAttributeNames": params["MessageAttributeNames"] + ["_datadog"]})
+
+                result = original_func(*args, **kwargs)
+                _set_response_metadata_tags(span, result)
+
+                try:
+                    if "Messages" in result:
+                        for message in result["Messages"]:
+                            max_attributes_flag = True
+                            if "MessageAttributes" in message:
+                                if "_datadog" in message["MessageAttributes"]:
+                                    if "StringValue" in message["MessageAttributes"]["_datadog"]:
+                                        max_attributes_flag = False
+                                        if PROPAGATION_KEY_BASE_64 in json.loads(
+                                                message["MessageAttributes"]["_datadog"]["StringValue"]
+                                        ):
+                                            pathway = json.loads(message["MessageAttributes"]["_datadog"]["StringValue"])[
+                                                PROPAGATION_KEY_BASE_64
+                                            ]
+                                            ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
+                                            ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
+
+                                        else:
+                                            log.warning("Unable to find data streams context for the message. Data "
```

Comment:
I feel like this should be a debug statement; otherwise if a queue has 100s or 1000s of messages, customesr will be spammed with WARNING messages.

Maybe @mabdinur has an opinion

Reply:
Yup debug log would be nice here. In the past we have gotten escalations for noisy error and warning logs.